src/jsdoc.ts

```diff
@@ -213,9 +220,20 @@ export function parseContents(commentText: string): ParsedJSDocComment|null {
         tagName = 'return';
       }
       let type: string|undefined;
-      if (JSDOC_TAGS_BLACKLIST.has(tagName)) {
-        warnings.push(`@${tagName} annotations are redundant with TypeScript equivalents`);
-        continue;  // Drop the tag so Closure won't process it.
+      if (JSDOC_TAGS_INPUT_BLACKLIST.has(tagName)) {
+        if (tagName !== 'template') {
+          // Tell the user to not write blacklisted tags, because there is TS
+          // syntax available for them.
+          warnings.push(`@${tagName} annotations are redundant with TypeScript equivalents`);
```

Comment:
does this warning have some other context when printed to the user?

Reply:
Yeah, it shows up via the TS api so it points at the relevant file/line etc.